src/faand.rs

```diff
@@ -192,17 +186,10 @@ pub(crate) async fn broadcast_first_send_second<
     n: usize,
     phase: &str,
     vec: &[Vec<(T, S)>],
-    len: usize,
 ) -> Result<Vec<Vec<(T, S)>>, Error> {
-    for k in (0..n).filter(|k| *k != i) {
-        send_to(channel, k, phase, &vec[k]).await?;
-    }
-
-    // 3 b) Receiving all commitments.
-    let mut recv_vec = vec![vec![]; n];
-    for k in (0..n).filter(|k| *k != i) {
-        recv_vec[k] = recv_vec_from::<(T, S)>(channel, k, phase, len).await?;
-    }
+    // TODO IMPOARTANT: Why is this method called broadcast but then does a scatter for both elements
+    //  and verifies the sending of the first vec? Why does this not fail the tests?
+    let recv_vec = scatter(channel, i, phase, vec).await?;
```

Comment:
@kisakishy Do you know why this method does a scatter (which it effectively already did before my change as each party is sent a different `vec[k]`) but then does the broadcast verification on the sent `T` values and it works?
Does `vec` maybe only differ in the `S` values?